src/analyzer/plugins/analyzerpluginsupport.cpp

```diff
@@ -0,0 +1,29 @@
+#include "analyzer/plugins/analyzerpluginsupport.h"
+
+QString AnalyzerPluginSupportInfo::matchOrGetDefaultPluginId(const QString& pluginIdToMatch) {
+    if (const auto plugins = availablePlugins(); !plugins.isEmpty()) {
+        if (std::any_of(std::begin(plugins), std::end(plugins), [&](const auto& info) {
+                return info.id() == pluginIdToMatch;
+            })) {
+            return pluginIdToMatch; // configured Plug-In available;
+        } else {
+            return defaultPlugin();
+        }
+    }
+    return m_pluginId;
```

Comment:
I struggle to understand this reasoning. Why do we return `m_pluginId` when no plugins are available but above we return the `defaultPlugin`?

Reply:
[This is a translation of the previous code (and its specifications).](#diff-4186bd7b4da09d30a81e5b021b8991aacb26f9e75310c37403199da822c506beL60-L70)

As the function `matchOrGetDefaultPluginId` assigns its results to a variable after returning, according to the old code, the function should return `m_pluginId` value if the first predicate-check fails.

As for _the design spec & functionality of the old code,_ well... I don't know about that specification when it was created... but that would be outside of the scope of this consolidation PR.

Reply:
Mhmm. I don't like the fact that we're returning `m_pluginId`. Yes, its (almost) the logical equivalent of what its replacing, but it doesn't match the function name and it also assumes that the function is used only when assigning to `m_pluginId`. I mean almost because if we're super pedantic about it, it doesn't specify `m_pluginId` is returned and assigned, it means `m_pluginId` is not being touched. Depending on the type, this makes a difference (though it probably doesn't since `m_pluginId` is just a (likely empty) Qstring).

My biggest problem with this function is, that its not obvious that it will return `m_pluginId` instead of the default when there are no plugins available. That is not obvious from the function definition and the reason is even less obvious when looking at the implementation.

Reply:
I totally agree with your analysis.

>  I mean almost because if we're super pedantic about it, it doesn't specify m_pluginId is returned and assigned, it means m_pluginId is not being touched. Depending on the type, this makes a difference (though it probably doesn't since m_pluginId is just a (likely empty) Qstring).

From the original old code-section, it was unclear what was happening with `m_pluginId` / `m_pluginId` purpose. I wanted to make the section "testable", and extracted the code portion into its own function. This makes the code (and its odd requirements / specifications) clearer, however, your point stands. Quietly, I am happy that it has brought into question (the code's spec and) further code-design discussions.

Again, this is outside of the scope of the consolidation PR, but having duplicated code, which was semi-unclear in its purpose, I felt that this rewrite at least clears up the code's spec a bit more (for all). I wanted to push these adjusted changes to mainline (as good practice) and further work on the code sections on a later date.

Reply:
> ...it means `m_pluginId` is not being touched. 

(This comment made me realise that the function was not marked as `const`. This has now been corrected).

Reply:
Thank very much you for clarifying. Even though I'm grateful for your contribution, my opinion stated below still stands. So consider my review dismissed.